nix/buildLeanPackage.nix

```diff
@@ -44,11 +59,27 @@ with builtins; let
     allowSubstitutes = false;
   };
   srcRoot = src;
-  allDeps = lib.foldr (dep: allDeps: allDeps // { ${dep.name} = dep; } // dep.allDeps) {} deps;
+
+  # A flattened list of Lean-module dependencies (`deps`)
+  allExternalDeps = lib.foldr (dep: allExternalDeps: allExternalDeps // { ${dep.name} = dep; } // dep.allExternalDeps) {} deps;
+  allNonBuiltInDeps = lib.filter (dep: !(builtInDepNames ? ${dep.name})) (attrValues allExternalDeps);
```

Comment:
Why do we need to filter out the stdlib?

Reply:
Fixed.

I tried to avoid modifying bootstrap.nix but that turns out to be easy so I just did that now.

Reply:
Nvm, this is more annoying than I though. Currently, IIUC leanc.in adds `-lStd -lInit -lLean` when building executables, but not when building shared libs. So if we don't filter them out, we'll be linking against the stdlib twice when building executables -- once from the leanc.in params, once from the Nix. 

Ended up removing the filtering code for simplicity -- linking twice is harmless other than a minimal slowdown. But fixing this properly would require some changes to the non-nix build setup as well, and I want to keep this PR small.

Reply:
Yeah, that's fine for now